creator-node/src/apiHelpers.js

```diff
@@ -10,13 +10,23 @@ const { generateTimestampAndSignature } = require('./apiSigning')
 module.exports.handleResponse = (func) => {
   return async function (req, res, next) {
     try {
+      const prometheusRegistry =
+        req.app.get('serviceRegistry').prometheusRegistry
+      const httpRequestTimerMetric = await prometheusRegistry.getMetricInstance(
+        'http_request_duration_seconds'
+      )
+      const endTimer = httpRequestTimerMetric.startTimer()
```

Comment:
Disclaimer: I don't like this pattern, even if the docs show how it's useful (and this could just be me learning Node's stylistic nuances).

In my PR, I create `const metric` and then use the `metric` object to record a metric by calling `metric.save()`.

In this PR, we call `startTimer()` which ideally would return a `Timer` object, but we call it an `EndTimer`. Then we call `EndTimer` like a function to finalize our actions on `const httpRequestTimerMetric`, while never even needing `httpRequestTimerMetric` as a `const`.

Additionally, could you try using a different file for your base example? Ideally a file that would lend itself nicely to measuring smaller chunks of work as part of a larger metric, similar to the usage of `metric` in [this file](https://github.com/AudiusProject/audius-protocol/blob/3fc3ddbee2389cee17e05894177ae96d08abf471/discovery-provider/src/tasks/index.py#L559).

Reply:
thanks for input here. few thoughts
- `metric.save()` doesn't exist in the js client, prob specific to python (not ideal here since python is an official prom client while the js one is unofficial - this is also very obvious by how poor the documentation is lol)
- `starttimer()` returns a function, not an object. when that function is called, the value is observed. per docs, this is provided as a [utility function](https://github.com/siimon/prom-client#utility-methods)
- i don't follow your point about `never even needing httpRequestTimerMetric as a const`
- yeah, i'll use a different file with a better example, good idea

what don't you like about this pattern? it's is the API provided by the client. it also makes sense to me (fwiw this pattern exists in python as well, and i'm not aware of any differences)
i looked through the python prometheus code, and the js equivalent of `.save()` is `.observe()` ([see docs](https://github.com/siimon/prom-client#examples)). this is to record a point-in-time value
we have 2 equivalent methods of recording histogram for duration:
```
// option 1
const metric = new Histogram(...route_duration...)
route function: {
  const start = Date.now()
  execute_logic()
  const end = Date.now()
  metric.observe(end)
}

// option 2
const metric = new Histogram(...route_duration...)
route function: {
  const endTimer = metric.startTimer()
  execute_logic()
  endTimer()
}
```

option2 seems like the more "correct" and readable way, wdyt? also cc @3166 for input

Reply:
Sure thing! A few notes:

* [metric.save()](https://github.com/AudiusProject/audius-protocol/blob/dbb1b3514c92fa7d521a461a3d0bc4e3d72978d0/discovery-provider/src/utils/prometheus_metric.py#L63) was added by me as a way to simplify our metrics interface. It was not part of the official client.
* Indeed on `startTimer()` returning a function. In Python, we'd typically name the left side to something like `end_timer_func` to denote that it's a function. My comment here was mainly me trying to work through the stylistic differences between the languages. :)
* I was under the assumption that we could combine lines 15 through 18 in one line, but missed that we have an `await` on line 15. Please disregard this comment.
* Major +1 on liking Option 2 over Option 1!

And now that I've had some time sitting with the code, I now no longer mind the style. It's not how I would have chosen, but it's:

* clean,
* makes sense,
* functional,
* and it's part of the official documentation.

I'm good with the way the timers are used in this PR! :)

Reply:
awesome!! as always, really appreciate you thinking through and discussing in detail and being open to changing your mind

glad you agree - and def agreed on naming like `end_timer_func` for clarity - i'll make that change!